epoch-indexer/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+pub async fn get_protocol_config(
+    epoch_start_height: u64,
+    client: &near_jsonrpc_client::JsonRpcClient,
+    try_another_blocks: bool,
+) -> anyhow::Result<near_chain_configs::ProtocolConfigView> {
+    let mut attempt_counter = 0;
+    let mut block_height = epoch_start_height;
+    loop {
+        let params =
+            near_jsonrpc_client::methods::EXPERIMENTAL_protocol_config::RpcProtocolConfigRequest {
+                block_reference: near_primitives::types::BlockReference::BlockId(
+                    near_indexer_primitives::types::BlockId::Height(block_height),
+                ),
+            };
+        match client.call(params).await {
+            Ok(response) => return Ok(response),
+            Err(e) => {
+                attempt_counter += 1;
+                if try_another_blocks {
+                    block_height += 1;
+                }
+                tracing::debug!(
+                    "Attempt: {}. Epoch_start_height {}. Error fetching protocol config: {:?}",
+                    attempt_counter,
+                    block_height,
+                    e
+                );
+                if attempt_counter > 20 {
+                    anyhow::bail!(
+                        "Failed to fetch epoch protocol config for epoch_start_height: {}",
+                        epoch_start_height
+                    )
+                }
+            }
+        }
+    }
+}
+
+pub async fn get_epoch_info_by_id(
+    epoch_id: CryptoHash,
+    rpc_client: &near_jsonrpc_client::JsonRpcClient,
+) -> anyhow::Result<readnode_primitives::IndexedEpochInfo> {
+    let validators_info = get_epoch_validators(epoch_id, rpc_client).await?;
+
+    let protocol_config =
+        match get_protocol_config(validators_info.epoch_start_height, rpc_client, false).await {
+            Ok(protocol_config) => protocol_config,
+            Err(e) => {
+                tracing::warn!(
+                    "Error fetching protocol config: {:?}. Try with another blocks",
+                    e
+                );
+                get_protocol_config(validators_info.epoch_start_height, rpc_client, true).await?
+            }
+        };
+
+    Ok(readnode_primitives::IndexedEpochInfo {
+        epoch_id,
+        epoch_height: validators_info.epoch_height,
+        epoch_start_height: validators_info.epoch_start_height,
+        validators_info,
+        protocol_config,
+    })
+}
+
+pub async fn get_epoch_info_by_block_height(
+    block_height: u64,
+    s3_client: &near_lake_framework::s3_fetchers::LakeS3Client,
+    s3_bucket_name: &str,
+    rpc_client: &near_jsonrpc_client::JsonRpcClient,
+) -> anyhow::Result<readnode_primitives::IndexedEpochInfo> {
+    let block_heights = near_lake_framework::s3_fetchers::list_block_heights(
+        s3_client,
+        s3_bucket_name,
+        block_height,
+    )
+    .await?;
+    let block = near_lake_framework::s3_fetchers::fetch_block_or_retry(
```

Comment:
Is there a reason to use S3 since we already utilize the Archival RPC?

Reply:
We seem to have agreed not to use the archived rpc. Isn't it?